### PR TITLE
feat: Add an option to limit the duration of starship directory scanning

### LINKED ISSUE #588: Slow to display prompt in directory with many (260k) files
#### Current Behavior
Prompt takes a long time (well, 2 seconds, which is a long time to wait for a prompt) in a directory with 264755 files

#### Expected Behavior
Prompt is speedy :)

#### Additional context/Screenshots
image

#### Environment
- Starship version: 0.25.1
- Shell type: bash
- Shell version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)
- Shell plugin manager: none
- Terminal emulator: GNOME Terminal 3.28.2 using VTE 0.52.2 +GNUTLS -PCRE2
- Operating system: ubuntu 18.04

#### Relevant Shell Configuration

```
eval "$(starship init bash)"
```

#### Starship Configuration
```
add_newline = false
[character]
symbol = "🤔"
```

### PR DESCRIPTION
**Pull request by heyrict**
#### Description
Starship will now only scan a subset of the current directory to avoid slow down in a folder with a large amount of files. Currently the default limit is 3000.

(The limit is set so that it will cost less than 50ms to render the prompt. 264755 / 2.715 * 0.05 \sim 4876)

#### Motivation and Context
Fixes #588

#### Screenshots

Screenshot from 2019-10-27 09-28-46

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by matchai** [comment]
I would personally prefer a timeout to a quantity of files. File count is a little arbitrary and may differ in impact when comparing systems.

Setting a 50ms timeout would mean the same thing on all systems. Thoughts?

**Comment by heyrict** [comment]
@matchai Done.

**Comment by Avi-D-coder** [comment]
Even with `scan_timeout=1` just running `ls` is very slow in the rust repo. I love the look of starship, but I have to switch back to powerlevel9k.

**Comment by heyrict** [comment]
Maybe we should also set a timeout to git_* modules in a separate PR. I can reproduce what @Avi-D-coder mentioned by:

1. Clone any nodejs project
2. Run `npm install`
3. Run `git add node_modules`

Thoughts? @matchai

File: docs/config/README.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ This is the list of prompt-wide configuration options.
 | -------------- | ----------------------------- | ------------------------------------------------------ |
 | `add_newline`  | `true`                        | Add a new line before the start of the prompt.         |
 | `prompt_order` | [link](#default-prompt-order) | Configure the order in which the prompt module occurs. |
+| `scan_timeout` | `30`                          | Timeout for starship to scan files (in milliseconds).  |
```
Comment (matchai) [inline docs/config/README.md]: 30 ms seems like a good default. 👍 
Testing locally on my 2.2 GHz i7 Macbook Pro 2018, the starship project was able to finish scanning in ~300µs, while a directory with 10,000 files took 21ms.

**Comment by Avi-D-coder** [comment]
@matchai  Anecdotally There has been a regression in the past week or so

**Pull request merged**